include/xtensor/xchunked_array.hpp

```diff
@@ -48,8 +45,36 @@ namespace xt
             m_chunks.resize(shape_chunk);
         }
 
+        reference operator[](const xindex& index)
+        {
+            reference el = element(index.cbegin(), index.cend());
+            return el;
+        }
+
+        const_reference operator[](const xindex& index) const
+        {
+            const_reference const_el = element(index.cbegin(), index.cend());
+            return const_el;
+        }
+
     private:
 
+        xt::xarray<chunk_type> m_chunks;
+        std::vector<size_t> m_shape;
+        std::vector<size_t> m_chunk_shape;
+
+        template <class... Idxs>
+        inline reference get_val(Idxs... idxs)
+        {
+            auto chunk_indexes_packed = get_chunk_indexes(std::make_index_sequence<sizeof...(Idxs)>(), idxs...);
+            auto chunk_indexes = unpack(chunk_indexes_packed);
+            auto indexes_of_chunk(std::get<0>(chunk_indexes));
+            auto indexes_in_chunk(std::get<1>(chunk_indexes));
+            chunk_type chunk = m_chunks.element(indexes_of_chunk.cbegin(), indexes_of_chunk.cend());
```

Comment:
I think this method could return a `std::pair<chunk, indexdex_in_chunk>`. This way, it can be const and called from both `operator()` and `operator() const` method.

Then you only need to "repeat" the return statement:
```cpp
return chunk.element(indexes_in_chunk.cbegin(), indexes_in_chunk.cend());
```